### PR TITLE
add verify email flow

### PR DESCRIPTION
**Pull request by santoshyadavdev**
The PR adds the functionality to verify email if not verified already.

We can revisit this feature, as I think this is not the correct way to implement as verification email will be sent every time an user logins with unverified email, I am thinking about showing a banner, which says your email is not verified, click here to receive verification email 

@mathisvester is it possible to add this banner in this PR or we can merge this and you can open a new PR

### CODE REVIEW

**Comment by mathisvester** [comment]
@santoshyadavdev while using the deploy preview of this PR and click on "Send Verification Email" after creating a new account I got an error:

```
{
    "message": "User (role: guests) missing scope (account)",
    "code": 401,
    "type": "general_unauthorized_scope",
    "version": "0.12.68"
}
```

It should be possible to add the banner in this PR. I think we have already some examples using https://material.angular.io/components/snack-bar. I think blocking login until the user has verified email would be the best suitable process. Is it similar to your idea?

To be sure after registering successfully we call the login endpoint (createEmailSession) and after that fetch the current user (account.get)? I think this process is crucial to not run in this _User (role: guests) missing scope (account)_ error.

File: libs/user/store/src/lib/store/user.actions.ts [hunk 1]
```diff
@@ -20,10 +21,11 @@ export const userActions = createActionGroup({
       message?: string;
     }>(),
     emailLoginFailure: props<{ error: HttpErrorResponse }>(),
-    creatJWTToken: emptyProps(),
-    createJWTTokenSuccess: props<{ token: string }>(),
-    createJWTTokenFailure: props<{ error: HttpErrorResponse }>(),
     logout: emptyProps(),
     getCurrentUserSuccess: props<{ user: User }>(),
+    SendEmailVerificationEmail: emptyProps(),
```
Comment (mathisvester) [inline libs/user/store/src/lib/store/user.actions.ts:26]: @santoshyadavdev is UpperCamelCase recommended? We should consider this in other action names.

File: libs/user/ui/src/lib/verify/verify.component.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<p>verify works!</p>
```
Comment (mathisvester) [inline libs/user/ui/src/lib/verify/verify.component.html:1]: @santoshyadavdev where is this component used?

File: libs/user/store/src/lib/store/user.actions.ts (hunk 1 above)
Comment (santoshyadavdev) [reply libs/user/store/src/lib/store/user.actions.ts:26]: My bad, changed.

**Comment by santoshyadavdev** [comment]
> @santoshyadavdev while using the deploy preview of this PR and click on "Send Verification Email" after creating a new account I got an error and no email in my inbox:
> 
> ```
> {
>     "message": "User (role: guests) missing scope (account)",
>     "code": 401,
>     "type": "general_unauthorized_scope",
>     "version": "0.12.68"
> }
> ```
> 
> It should be possible to add the banner in this PR. I think we have already some examples using https://material.angular.io/components/snack-bar. I think blocking login until the user has verified email would be the best suitable process. Is it similar to your idea?
> 
> To be sure after registering successfully we call the login endpoint (createEmailSession) and after that fetch the current user (account.get)? I think this process is crucial to not run in this _User (role: guests) missing scope (account)_ error.

Sorry looks like I pushed some incomplete code, you can check now, the idea is to redirect to login after registration.

You can verify now, it should work as expected, The verify component is only used for verification flow.
The user will receive an email with below link and when user clicks on it and it will redirect to verify component.
Screenshot 2024-02-11 at 14 41 09

**Comment by mathisvester** [comment]
@santoshyadavdev still got an error when clicking on "Send Verification Email" and received no email.

```
{
    "message": "Invalid `url` param: URL host must be one of: localhost, cloud.appwrite.io, appwrite.io, *.netlify.app",
    "code": 400,
    "type": "general_argument_invalid",
    "version": "0.12.72"
}
```

And I have some additional ideas:
- Can we send the verification directly after registering and show a banner: "Registration was successful! Please verify your account (...)" instead of redirecting to login?
- If the user clicks on the link in the email can we create a magic link session so that that the user will be logged in automatically?
- If the users logs in and has not verified can we stop the process (until verified) and show a banner "Account not verified. _Resend_ verification email"?

**Comment by santoshyadavdev** [comment]
> @santoshyadavdev still got an error when clicking on "Send Verification Email" and received no email.
> 
> ```
> {
>     "message": "Invalid `url` param: URL host must be one of: localhost, cloud.appwrite.io, appwrite.io, *.netlify.app",
>     "code": 400,
>     "type": "general_argument_invalid",
>     "version": "0.12.72"
> }
> ```
> 
> And I have some additional ideas:
> 
> * Can we send the verification directly after registering and show a banner: "Registration was successful! Please verify your account (...)" instead of redirecting to login?
> * If the user clicks on the link in the email can we create a magic link session so that that the user will be logged in automatically?
> * If the users logs in and has not verified can we stop the process (until verified) and show a banner "Account not verified. **Resend** verification email"?

looks like the this custom-deploy url generated is not a valid url for appwrite.

- We can do the login after registration, thats doable, i can do that in follow-up PR
- the email sent is for verifying token, I think there is another API call for magic link.
- I will change the message.

**Review by mathisvester — APPROVED** [review]

**Comment by mathisvester** [comment]
@santoshyadavdev sound good for me.

**Pull request merged**